### ISSUE TITLE
should lattice_fold/lattice_reduce be replaced by modules?

### ISSUE THREAD

**Issue by zzlk** [hydro-project/hydroflow#904]
The main issue is that fold/lattice_fold have different flow props.

**Comment by jhellerstein** [comment]
`lattice_fold` is a special case of `fold` -- the lattice merge function is monotone and hence shouldn't need coordination.

Should we have `monotonic_streaming_fold` as a thing?

In principle `lattice_fold` is monotone, but the Q is whether it outputs a stream or a singleton. If a stream, that's a monotone stream and no need to seal/stratify. If it's a singleton, need to stratify. Probably should be a stream, but we need to figure out how to track all the stream/cumulative stuff as in #927

**Comment by MingweiSamuel** [comment]
Nope, HF+, also cleaning up those operators for state externalization #1058
